activerecord/lib/active_record.rb

```diff
@@ -196,6 +196,32 @@ def self.default_timezone=(default_timezone)
 
   self.default_timezone = :utc
 
+  # The action to take when database query produces warning.
+  # Must be one of :ignore, :log, :raise, :report, or a custom proc.
+  # The default is :ignore.
+  singleton_class.attr_reader :db_warnings_action
+
+  def self.db_warnings_action=(action)
+    @db_warnings_action = case action
+    when :ignore
+      ->(_, _) {}
+    when :log
+      ->(warning, _) { logger.warn("[#{warning.class}] #{warning.message} (#{warning.code})") }
+    when :raise
+      ->(warning, _) { raise warning }
+    when :report
+      ->(warning, _) { Rails.error.report(warning, handled: true) }
```

Comment:
Wondering if it's worth setting anything from the `db_config` in `context` -- like maybe the connection name?